java/src/main/java/com/google/crypto/tink/KmsClient.java

```diff
@@ -27,6 +27,11 @@
   /** @return true if this client does support {@code keyUri} */
   public boolean doesSupport(String keyUri);
 
+  /**
+   * @param keyUri the keyUri to set
+   */
+  public void setKeyUri(String keyUri);
```

Comment:
Hi there,

Adding a new method to a public interface is a backward-incompatibility change, so in general we don't want to do this unless there's no other option.

To make withCredentialsProvider() work, we can move the code that builds an AWS client from withCredentialsProvider() to getAead(). See my comment below.

Reply:
Thanks for your suggestion. your patch makes sense. May I rewrite this PR as you wrote?

Reply:
Sorry for the slow response. Yes, please rewrite the PR.